### DISCUSSION TITLE
Is it possible to display e..g. AmazonPrime or Youtube on the car display? Other than screen-mirroring?

### DISCUSSION THREAD

**Discussion by wanderschikanentv** [BimmerGestalt/AAIdrive#641] [Q&A]
Hi,
I'm a happy user of AAIdrive in an F20 1-series with Navigation Professional from 01-2019. So far, I use Spotify and ARD Audiothek, they both work well.

I'm now interested whether I can also display video on the car display, from Youtube or Amazon Prime or e.g. ARD or ZDF Mediathek or VLC. I know that one way is to use the Screen Mirroring, but is there also a different, direct way to get the video on the car display?

I'm asking because I can select Amazon Prime, Youtube and VLC as music apps in AAIdrive, but I don't understand whether or how this can stream the video.

WBR,
Mike

**Comment by hufman** [comment]
Thank you for the compliments!
As far as I know, there isn't any general way for an app to get another app's video stream. The MediaController method that AAIdrive uses to control music apps only offers metadata and control methods. There isn't any way to embed apps within another app for background interaction.
Android Auto has a special integration where it passes a Surface object to a private API in each app for the app to render to, but that API is protected and AAIdrive can't use it.
I also can't encourage watching videos while driving :)

**Reply by wanderschikanentv** [reply]
Thank you for your feedback! OK, understood, I thought it may be possible to stream video, maybe passing it in a similar way like you can watch videos which are stored on an USB flashdrive.

And finally, of course, I appreciate your attitude of encouraging safe driving!
